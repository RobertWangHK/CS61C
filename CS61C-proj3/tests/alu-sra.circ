<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project source="2.7.1" version="1.0">
This file is intended to be loaded by Logisim (http://www.cburch.com/logisim/).
<lib desc="#Wiring" name="0">
    <tool name="Splitter">
      <a name="facing" val="west"/>
      <a name="incoming" val="32"/>
      <a name="appear" val="center"/>
      <a name="bit1" val="0"/>
      <a name="bit2" val="0"/>
      <a name="bit3" val="0"/>
      <a name="bit4" val="0"/>
      <a name="bit5" val="0"/>
      <a name="bit6" val="0"/>
      <a name="bit7" val="0"/>
      <a name="bit8" val="0"/>
      <a name="bit9" val="0"/>
      <a name="bit10" val="0"/>
      <a name="bit11" val="0"/>
      <a name="bit12" val="0"/>
      <a name="bit13" val="0"/>
      <a name="bit14" val="0"/>
      <a name="bit15" val="0"/>
      <a name="bit16" val="1"/>
      <a name="bit17" val="1"/>
      <a name="bit18" val="1"/>
      <a name="bit19" val="1"/>
      <a name="bit20" val="1"/>
      <a name="bit21" val="1"/>
      <a name="bit22" val="1"/>
      <a name="bit23" val="1"/>
      <a name="bit24" val="1"/>
      <a name="bit25" val="1"/>
      <a name="bit26" val="1"/>
      <a name="bit27" val="1"/>
      <a name="bit28" val="1"/>
      <a name="bit29" val="1"/>
      <a name="bit30" val="1"/>
      <a name="bit31" val="1"/>
    </tool>
    <tool name="Pin">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Probe">
      <a name="radix" val="10signed"/>
    </tool>
    <tool name="Tunnel">
      <a name="width" val="32"/>
    </tool>
    <tool name="Pull Resistor">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Clock">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Constant">
      <a name="value" val="0x0"/>
    </tool>
    <tool name="Power">
      <a name="facing" val="south"/>
    </tool>
  </lib>
  <lib desc="#Gates" name="1">
    <tool name="NOT Gate">
      <a name="facing" val="south"/>
    </tool>
    <tool name="Buffer">
      <a name="width" val="3"/>
    </tool>
    <tool name="AND Gate">
      <a name="width" val="16"/>
      <a name="inputs" val="2"/>
    </tool>
    <tool name="OR Gate">
      <a name="inputs" val="2"/>
    </tool>
    <tool name="NOR Gate">
      <a name="inputs" val="2"/>
    </tool>
    <tool name="XOR Gate">
      <a name="inputs" val="2"/>
    </tool>
    <tool name="Odd Parity">
      <a name="facing" val="south"/>
      <a name="inputs" val="3"/>
    </tool>
  </lib>
  <lib desc="#Plexers" name="2">
    <tool name="Multiplexer">
      <a name="select" val="3"/>
    </tool>
  </lib>
  <lib desc="#Arithmetic" name="3">
    <tool name="Adder">
      <a name="width" val="16"/>
    </tool>
    <tool name="Subtractor">
      <a name="width" val="16"/>
    </tool>
    <tool name="Multiplier">
      <a name="width" val="1"/>
    </tool>
    <tool name="Divider">
      <a name="width" val="16"/>
    </tool>
    <tool name="Negator">
      <a name="width" val="1"/>
    </tool>
    <tool name="Comparator">
      <a name="width" val="16"/>
    </tool>
  </lib>
  <lib desc="#Memory" name="4">
    <tool name="Register">
      <a name="width" val="16"/>
    </tool>
    <tool name="RAM">
      <a name="dataWidth" val="16"/>
    </tool>
    <tool name="ROM">
      <a name="contents">addr/data: 8 8
0
</a>
    </tool>
  </lib>
  <lib desc="#I/O" name="5"/>
  <lib desc="#Base" name="6">
    <tool name="Text Tool">
      <a name="text" val=""/>
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="valign" val="base"/>
    </tool>
  </lib>
  <lib desc="file#alu.circ" name="7"/>
  <main name="main"/>
  <options>
    <a name="gateUndefined" val="ignore"/>
    <a name="simlimit" val="1000"/>
    <a name="simrand" val="0"/>
  </options>
  <mappings>
    <tool lib="6" map="Button2" name="Menu Tool"/>
    <tool lib="6" map="Ctrl Button1" name="Menu Tool"/>
    <tool lib="6" map="Button3" name="Menu Tool"/>
  </mappings>
  <toolbar>
    <tool lib="6" name="Poke Tool"/>
    <tool lib="6" name="Edit Tool"/>
    <tool lib="6" name="Text Tool">
      <a name="text" val=""/>
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="valign" val="base"/>
    </tool>
    <sep/>
    <tool lib="0" name="Pin">
      <a name="facing" val="west"/>
      <a name="width" val="16"/>
      <a name="tristate" val="false"/>
    </tool>
    <tool lib="0" name="Pin">
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
      <a name="label" val="Result"/>
      <a name="labelloc" val="east"/>
    </tool>
    <tool lib="1" name="NOT Gate"/>
    <tool lib="1" name="AND Gate"/>
    <tool lib="1" name="OR Gate"/>
  </toolbar>
  <circuit name="main">
    <a name="circuit" val="main"/>
    <a name="clabel" val=""/>
    <a name="clabelup" val="east"/>
    <a name="clabelfont" val="SansSerif plain 12"/>
    <wire from="(180,30)" to="(200,30)"/>
    <wire from="(40,100)" to="(40,160)"/>
    <wire from="(80,160)" to="(90,160)"/>
    <wire from="(150,100)" to="(150,130)"/>
    <wire from="(90,140)" to="(100,140)"/>
    <wire from="(100,40)" to="(100,50)"/>
    <wire from="(330,190)" to="(370,190)"/>
    <wire from="(180,110)" to="(190,110)"/>
    <wire from="(240,40)" to="(260,40)"/>
    <wire from="(390,80)" to="(570,80)"/>
    <wire from="(60,180)" to="(60,200)"/>
    <wire from="(180,190)" to="(190,190)"/>
    <wire from="(430,150)" to="(630,150)"/>
    <wire from="(40,160)" to="(50,160)"/>
    <wire from="(170,330)" to="(190,330)"/>
    <wire from="(170,50)" to="(170,160)"/>
    <wire from="(100,50)" to="(170,50)"/>
    <wire from="(180,110)" to="(180,160)"/>
    <wire from="(390,200)" to="(390,330)"/>
    <wire from="(170,160)" to="(180,160)"/>
    <wire from="(180,160)" to="(180,190)"/>
    <wire from="(90,160)" to="(170,160)"/>
    <wire from="(410,110)" to="(570,110)"/>
    <wire from="(140,130)" to="(150,130)"/>
    <wire from="(90,120)" to="(100,120)"/>
    <wire from="(330,330)" to="(390,330)"/>
    <wire from="(390,80)" to="(390,100)"/>
    <wire from="(170,160)" to="(170,330)"/>
    <wire from="(170,50)" to="(200,50)"/>
    <wire from="(90,140)" to="(90,160)"/>
    <wire from="(330,110)" to="(370,110)"/>
    <wire from="(40,100)" to="(150,100)"/>
    <comp lib="0" loc="(100,40)" name="Pin">
      <a name="facing" val="south"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
      <a name="label" val="Test #"/>
      <a name="labelloc" val="north"/>
    </comp>
    <comp lib="0" loc="(90,120)" name="Constant">
      <a name="width" val="8"/>
    </comp>
    <comp lib="7" loc="(400,150)" name="main"/>
    <comp lib="4" loc="(330,190)" name="ROM">
      <a name="dataWidth" val="32"/>
      <a name="contents">addr/data: 8 32
deadbeef f0000000 0 feedf00d
</a>
    </comp>
    <comp lib="0" loc="(60,200)" name="Clock">
      <a name="facing" val="north"/>
    </comp>
    <comp lib="0" loc="(630,150)" name="Pin">
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
      <a name="width" val="32"/>
      <a name="label" val="Result"/>
      <a name="labelloc" val="east"/>
    </comp>
    <comp lib="4" loc="(330,110)" name="ROM">
      <a name="dataWidth" val="32"/>
      <a name="contents">addr/data: 8 32
22 12 22
</a>
    </comp>
    <comp lib="0" loc="(570,110)" name="Pin">
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
      <a name="label" val="Equal"/>
      <a name="labelloc" val="east"/>
    </comp>
    <comp lib="4" loc="(330,330)" name="ROM">
      <a name="dataWidth" val="4"/>
      <a name="contents">addr/data: 8 4
32*2
</a>
    </comp>
    <comp lib="3" loc="(240,40)" name="Comparator">
      <a name="mode" val="unsigned"/>
    </comp>
    <comp lib="6" loc="(412,64)" name="Text">
      <a name="text" val="YOUR ALU SHOULD FIT HERE"/>
    </comp>
    <comp lib="3" loc="(140,130)" name="Adder"/>
    <comp lib="0" loc="(180,30)" name="Constant">
      <a name="width" val="8"/>
      <a name="value" val="0xf"/>
    </comp>
    <comp lib="6" loc="(443,407)" name="Text">
      <a name="text" val="There should be no blue/red wires here, and the result shouldn't be x's."/>
    </comp>
    <comp lib="0" loc="(570,80)" name="Pin">
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
      <a name="label" val="Signed Overflow"/>
      <a name="labelloc" val="east"/>
    </comp>
    <comp lib="4" loc="(80,160)" name="Register"/>
    <comp lib="0" loc="(260,40)" name="Pin">
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
      <a name="label" val="halt"/>
      <a name="labelloc" val="east"/>
    </comp>
  </circuit>
</project>
