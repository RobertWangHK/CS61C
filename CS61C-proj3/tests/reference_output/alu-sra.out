0000 0000	0	0	1111 0111 1010 1011 0110 1111 1011 1011
0000 0001	0	0	1111 1111 1111 1111 1111 1100 0000 0000
0000 0010	0	0	0000 0000 0000 0000 0000 0000 0000 0000
0000 0011	0	0	1111 1110 1110 1101 1111 0000 0000 1101
0000 0100	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 0101	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 0110	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 0111	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1000	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1001	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1010	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1011	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1100	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1101	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1110	0	1	0000 0000 0000 0000 0000 0000 0000 0000
0000 1111	0	1	0000 0000 0000 0000 0000 0000 0000 0000
